scalyr_agent/scalyr_logging.py

```diff
@@ -175,7 +175,7 @@ def alternateCurrentFrame():
 
 
 # noinspection PyPep8Naming
-class AgentLogger(logging.Logger):
+class AgentLogger(logging.Logger, object):
```

Comment:
Does logger not already inherit from object?

Looking through the 2.7 source, it would appear to go `object -> logging.Filterer -> logging.Logger -> AgentLogger`

Reply:
I did see that and was initially confused too.  But when run on 2.6 it fails (not 2.7) . So it has to be the case that one of the parent classes in 2.6 fails to inherit object.

Reply:
You're right. I just looked through the 2.6 source, and yes, it turns out `Filterer` doesn't inherit from `object`.

What problem were you encountering without inheriting from object and what happens in python if the same parent class exists at multiple points in the inheritence hierarchy?

I probably wouldn't be asking this if it was happening on a class in the test library, but AgentLogger is used everywhere, so just want to make sure there aren't going to be any side-effects in 2.7, which presumably is the common use case for all our clients.

Also, if it's just a testing thing, is it possible to fix the problem in 2.6 by having the `FakeAgentLogger` inherit from object instead, therefore not affecting `AgentLogger` at all?

Reply:
This is a good thought. 

I wouldn't _think_ there'd be problems with repeat inheritance but definitely needs to be better understood. 

Lemme fire up 2.6 and look more closely again at whether it's possible to isolate this to only test code.

Reply:
> I wouldn't think there'd be problems with repeat inheritance but definitely needs to be better understood.

That would be my assumption too, but I don't know enough about the details to be sure.  If we can isolate it to test only code that'd be the easiest way to deal with this for now.

Reply:
Thanks again for this observation/catch.   Was able to move to `FakeAgentLogger` . btw -- the error happens when calling superclass __init__

![image](https://user-images.githubusercontent.com/48775713/56717261-19f2ba00-66f1-11e9-918f-4d6dc47186c8.png)

Reply:
That makes sense.  `super` only works with classes derived from `object`.

All looks good.  I'll approve this in a minute.